### PR TITLE
Fix tiup update --self with offline package

### PR DESCRIPTION
**Pull request by lucklove**
Fix https://github.com/pingcap/tiup/issues/658

Signed-off-by: lucklove <gnu.crazier@gmail.com>

### CODE REVIEW

File: pkg/repository/clone_mirror.go [hunk 1]
```diff
@@ -306,27 +306,6 @@ func cloneComponents(repo *V1Repository,
 		compManifests[name] = newManifest
 	}
 
-	// Download TiUP binary
-	for _, goos := range options.OSs {
```
Comment (lonng) [inline pkg/repository/clone_mirror.go:310]: Why delete this part?

Comment (lucklove) [reply pkg/repository/clone_mirror.go:310]: It's of no use in V1. This part doesn't change add `tiup` to manifests. But with V1 repository, tiup update self only read from manifests.

File: pkg/environment/env.go [hunk 2]
```diff
@@ -215,7 +215,17 @@ func (env *Environment) PlatformString() string {
...
+			return err
+		}
+
+		// Cover the root.json from tiup.bar.gz
+		profile := localdata.InitProfile()
+		if err := profile.ResetMirror(Mirror(), ""); err != nil {
+			return err
+		}
+
+		return nil
```
Comment (lonng) [inline pkg/environment/env.go:228]: The +224 to +228 can be simplified to `return profile.ResetMirror(Mirror(), "")`

**Comment by lonng** [comment]
@lucklove Please write the release note for this PR.

**Pull request merged**